tests/MongoDB.Driver.Tests/Specifications/Runner/MongoClientJsonDrivenTestRunnerBase.cs

```diff
@@ -356,10 +349,29 @@ protected void SetupAndRunTest(JsonDrivenTestCase testCase)
             SetupAndRunTest(testCase.Shared, testCase.Test);
         }
 
-        protected void VerifyCollectionData(IEnumerable<BsonDocument> expectedDocuments, Action<BsonDocument, BsonDocument> prepareExpectedResult)
+        protected virtual void VerifyCollectionData(
+            IEnumerable<BsonDocument> expectedDocuments,
+            string databaseName,
+            string collectionName,
+            ReadConcern readConcern = null)
+        {
+            VerifyCollectionData(expectedDocuments, databaseName, collectionName, readConcern: readConcern, prepareExpectedResult: null);
```

Comment:
The overrides of VerifyCollectionData should be:

        protected virtual void VerifyCollectionData(IEnumerable<BsonDocument> expectedDocuments)
        {
            VerifyCollectionData(expectedDocuments, DatabaseName, CollectionName);
        }

        protected virtual void VerifyCollectionData(IEnumerable<BsonDocument> expectedDocuments, string databaseName, string collectionName)
        {
            VerifyCollectionData(expectedDocuments, databaseName, collectionName, prepareExpectedResult: null);
        }

The second one is new.

Reply:
Maybe **not** move this method in the file so that the diff doesn't make it look like they are **both** new.